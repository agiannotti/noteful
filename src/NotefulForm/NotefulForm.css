.Noteful-form {
  background-color: #1d3656;
  color: #dedede;
  padding: 16px;
  max-width: 332px;
}

.Noteful-form label {
  font-size: 16px;
  padding-bottom: 8px;
}

.Noteful-form input,
.Noteful-form textarea,
.Noteful-form select {
  font-size: 14px;
  padding: 8px;
  width: 300px;
}

.Noteful-form select {
  height: 33px;
}

.Noteful-form > .field {
  margin-bottom: 16px;
}
.Noteful-form > .buttons {
  margin-top: 26px;
}

.Noteful-form button {
  background-color: #010d1e;
  border-radius: 4px;
  border: 0;
  color: #cacaca;
  display: block;
  font-size: 18px;
  font-weight: bold;
  padding: 8px;
  text-align: center;
  text-decoration: none;
}
.Noteful-form button[type="submit"]:hover {
  background-color: #b18d8b;
  color: white;
}
