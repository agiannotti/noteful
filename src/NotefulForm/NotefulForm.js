import React from 'react'
import './NotefulForm.css'

export default function NotefulForm(props) {
  const { className, ...otherProps } = props
  return (
    <form
      className={['Noteful-form', className].join(' ')}
      action='#'
      {...otherProps}
    />
  )
}
