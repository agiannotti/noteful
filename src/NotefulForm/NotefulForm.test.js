import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NotefulForm from './NotefulForm'

describe(`NotefulForm component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  }

  it('renders a form.NotefulForm by default', () => {
    const wrapper = shallow(<NotefulForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the NotefulForm given props', () => {
    const wrapper = shallow(<NotefulForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
