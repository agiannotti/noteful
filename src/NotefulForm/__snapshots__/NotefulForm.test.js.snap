// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotefulForm component renders a form.NotefulForm by default 1`] = `
<form
  action="#"
  className="Noteful-form "
/>
`;

exports[`NotefulForm component renders the NotefulForm given props 1`] = `
<form
  action="#"
  className="Noteful-form test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</form>
`;
